### PR TITLE
add profile

### PR DESCRIPTION
**Pull request by AlexeyDunnikov**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -62,6 +62,7 @@ app.use(varMiddleware(connection).basketAmount);
 app.use(varMiddleware(connection).categoriesList);
 
 //Routes
+require("./routes/pagesRoute")(app, connection);
```
Comment (vitali-working) [inline app.js]: `pagesRoute` can you add more specific name?

File: controllers/carsController.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ module.exports = (connection) => {
   const controllerMethods = {};
 
   controllerMethods.addCarModification = async (req, res) => {
+
```
Comment (vitali-working) [inline controllers/carsController.js]: why empty line? I see in file below in the same situation you have removed empty line. Be consistent

File: controllers/catalogController.js [hunk 3]
```diff
@@ -115,5 +114,28 @@ module.exports = (connection) => {
     res.render("catalog", options);
   };
 
+  controllerMethods.renderPart = async (req, res) => {
+    const partId = req.params.id;
+
```
Comment (vitali-working) [inline controllers/catalogController.js:119]: double empty lines

File: controllers/catalogController.js [hunk 4]
```diff
@@ -115,5 +114,28 @@ module.exports = (connection) => {
...
+    
+    let part; 
+    if (req.user) {
+      part = await catalogModel.getPartInfoByIdAndUser(
```
Comment (vitali-working) [inline controllers/catalogController.js:122]: Are you sure that `catalogModel` should know about current user?

File: controllers/indexController.js [hunk 5]
```diff
@@ -11,26 +11,23 @@ module.exports = (connection) => {
   const controllerMethods = {};
 
   controllerMethods.render = async (req, res) => {
-    const cars = await carsModel.getAllCars(req, res);
+    const cars = await carsModel.getAllCars();
+    const allCategories = await catalogModel.getCategories();
```
Comment (vitali-working) [inline controllers/indexController.js]: `getAllCars`, `getAllCategories` or  `getCars`, `getCategories`

File: controllers/indexController.js [hunk 6]
```diff
@@ -11,26 +11,23 @@ module.exports = (connection) => {
...
     if (req.user) {
-      const carsModificationsId =
-        await carsModel.getCarsModFromGarageWhereUserId(req.user.id);
-
-      for (const modId of carsModificationsId) {
-        const car = await carsModel.getModificationInfo(modId);
-        garage.push(car);
-      }
-      options.garage = garage.reverse();
+      const cars = await carsModel.getCarsInfoFromGarageByUser(req.user.id);
```
Comment (vitali-working) [inline controllers/indexController.js:28]: The same situation with `catalogModel`, `carsModel` shouldn't know about user

File: controllers/indexController.js [hunk 7]
```diff
@@ -11,26 +11,23 @@ module.exports = (connection) => {
...
+
+      options.garage = cars.reverse();
```
Comment (vitali-working) [inline controllers/indexController.js:30]: Why  is `reverse`?

File: controllers/userController.js [hunk 8]
```diff
@@ -22,6 +27,55 @@ module.exports = (connection) => {
     });
   };
 
+  controllerMethods.renderProfile = async (req, res) => {
```
Comment (vitali-working) [inline controllers/userController.js:30]: big function, please simplify it

File: controllers/catalogController.js (hunk 4 above)
Comment (AlexeyDunnikov) [reply controllers/catalogController.js:122]: I should know if this part in basket or no

File: controllers/indexController.js (hunk 6 above)
Comment (AlexeyDunnikov) [reply controllers/indexController.js:28]: When I render index page, I see if user authorize or no.
If user authorize, I take cars from garage (DB) and I need to know his id, but if he is not authorize I take cars from local storage.

File: controllers/indexController.js (hunk 7 above)
Comment (AlexeyDunnikov) [reply controllers/indexController.js:30]: Last car that we add to garage should be first in the list of cars in the page.

File: models/carsModel.js [hunk 9]
```diff
@@ -126,8 +126,8 @@ module.exports = (connection) => {
       connection.query(
         `SELECT modifications.id AS id, marks.name AS mark, models.name AS model, gen.name AS generation, gen.img AS img, gen.year_from AS year_from, gen.year_to AS year_to, engines.name AS engine_name, engines.power AS engine_power, engines.horses AS engine_horses, engines.value AS engine_value
         FROM car_marks AS marks 
-        INNER JOIN car_models AS models ON marks.id = models.car_id
-        INNER JOIN car_generations AS gen ON models.id = gen.model_id
+        INNER JOIN car_models AS models ON marks.id = models.id_car
```
Comment (vitali-working) [inline models/carsModel.js:129]: Why did you change naming? I guess `car_id` is much better than `id_car`

File: controllers/catalogController.js (hunk 4 above)
Comment (vitali-working) [reply controllers/catalogController.js:122]: `catalog` is not the same as `basket`. Catalog information should not depend on user.
